lib/mix/lib/mix/tasks/app.start.ex

```diff
@@ -105,4 +109,19 @@ defmodule Mix.Tasks.App.Start do
       true -> :temporary
     end
   end
+
+  defp check_configured() do
+    started = Mix.State.get(:started_application, [])
+    included = Enum.flat_map(started, fn(app) ->
+      {:ok, included} = :application.get_key(app, :included_applications)
+      included
+    end)
+    configured = Mix.State.get(:configured_application, [])
+    _ = for app <- configured -- ([:elixir, :mix, :iex] ++ included ++ started) do
+      Mix.shell.info "Configuration persisted for application #{app} but it was not started by mix. " <>
```

Comment:
About the error message:

```
You have configured application #{inspect app} but it was not started by mix.
This may mean two things:

  1. you (or a dependency) depend on application #{inspect app} but you haven't 
     listed it as a dependency in :applications (or :included_applications) in the
     mix.exs file

  2. you are configuring an application that does not really exist. Please ensure it
     exists or remove the configuration
```

Note: we don't use backticks when quoting code in error messages in Elixir. There is particular reason for not using them but we should be consistent here. :)